File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android/com/example/smoke/NoCtorsInitializedPublicFieldOneInternalField.java
/*

 *
 */

package com.example.smoke;

import android.support.annotation.NonNull;

/**
 * <p>Test case 2
 * <p>Input:
 * <ul>
 * <li>This structure does not define explicit constructors / field constructors.</li>
 * <li>It has one public field with default value.</li>
 * <li>One of fields is internal.</li>
 * </ul>
 * <p>Expected outcome:
 * <ul>
 * <li>Generated initialized args constructor is not public (because it would expose internal field).</li>
 * </ul>
 */
public final class NoCtorsInitializedPublicFieldOneInternalField {
    @NonNull
    String internalField;
    public boolean publicField;
    public int initializedField;

    NoCtorsInitializedPublicFieldOneInternalField(@NonNull final String internalField, final boolean publicField) {
        this.internalField = internalField;
        this.publicField = publicField;
        this.initializedField = 77;
    }


}

File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android-kotlin/com/example/smoke/NoCtorsInitializedPublicFieldOneInternalField.kt
/*

 *
 */

@file:JvmName("NoCtorsInitializedPublicFieldOneInternalFieldExtensions")


package com.example.smoke


/**
 * Test case 2
 *
 * Input:
 * - This structure does not define explicit constructors / field constructors.
 * - It has one public field with default value.
 * - One of fields is internal.
 *
 * Expected outcome:
 * - Generated initialized args constructor is not public (because it would expose internal field).
 */
class NoCtorsInitializedPublicFieldOneInternalField {
    @JvmField @JvmSynthetic internal var internalField: String
    @JvmField var publicField: Boolean
    @JvmField var initializedField: Int



    internal constructor(internalField: String, publicField: Boolean) {
        this.internalField = internalField
        this.publicField = publicField
        this.initializedField = 77
    }





}

